spec/lib/two_factor_authentication/models/two_factor_authenticatable_spec.rb

```diff
@@ -90,66 +133,67 @@ def instance.send_two_factor_authentication_code
   end
 
   describe '#provisioning_uri' do
+
     shared_examples 'provisioning_uri' do |instance|
-      before do
-        instance.email = 'houdini@example.com'
-        instance.run_callbacks :create
+      it 'fails until create_otp_secret is called' do
+        expect { instance.provisioning_uri }.to raise_error(Exception)
       end
 
-      it "returns uri with user's email" do
-        expect(instance.provisioning_uri).
-          to match(%r{otpauth://totp/houdini@example.com\?secret=\w{16}})
-      end
+      describe 'with secret set' do
+        before do
+          instance.create_otp_secret
+        end
 
-      it 'returns uri with issuer option' do
-        expect(instance.provisioning_uri('houdini')).
-          to match(%r{otpauth://totp/houdini\?secret=\w{16}$})
-      end
+        before do
+          instance.email = 'houdini@example.com'
+          instance.run_callbacks :create
```

Comment:
I don't think we need this anymore now that you have removed the `before_create` callback.